docs/go/how-to-create-a-worker-session-in-go.md

```diff
@@ -0,0 +1,122 @@
+  // ...
+  sessionOptions := &workflow.SessionOptions{
+    CreationTimeout:  time.Minute,
+    ExecutionTimeout: time.Minute,
+  }
+  sessionCtx, err := workflow.CreateSession(ctx, sessionOptions)
+  if err != nil {
+    return err
+  }
+  defer workflow.CompleteSession(sessionCtx)
+
+  var activityResult1 *ActivityResult1
+  err = workflow.ExecuteActivity(sessionCtx, ActivityFunction1, param1).Get(sessionCtx, &activityResult1)
+  if err != nil {
+    return err
+  }
+
+  var activityResult2 *ActivityResult2
+  err = workflow.ExecuteActivity(sessionCtx, ActivityFunction2, *activityResult1).Get(sessionCtx, &activityResult2)
+  if err != nil {
+    return err
+  }
+
+  return workflow.ExecuteActivity(sessionCtx, uploadFileActivityName, *activityResult2).Get(sessionCtx, nil)
+}
+```
+
+hen the `CreateSession` API is called, the Task queue name that is specified in the `ActivityOptions` is used (or in the `StartWorkflowOptions` if the Task Queue name is not specified in `ActivityOptions`), and a Session is created with one of the Workers polling that Task Queue.
+
+The Session Context is cancelled if the Worker executing this Session dies or `CompleteSession()` is called.
+When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error may be returned if the Session framework detects that the Worker executing this Session has died.
+The failure of Activity Executions won't affect the state of the Session, so you still need to handle the errors returned from your Activities and call `CompleteSession()` if necessary.
+
+`CreateSession()` will return an error if the context passed in already contains an open Session.
+If all the Workers are currently busy and unable to handle new Session, the framework will keep retrying until the `CreationTimeout` you specified in `SessionOptions` has passed before returning an error (check the **Concurrent Session Limitation** section for more details).
+
+`CompleteSession()` releases the resources reserved on the Worker, so it's important to call it as soon as you no longer need the Session.
+It will cancel the session context and therefore all the Activity Executions using that Session Context.
+Note that it's safe to call `CompleteSession()` on a failed Session, meaning that you can call it from a `defer` function after the Session is successfully created.
+
+### Session Metadata
+
+```go
+type SessionInfo struct {
+  // A unique Id for the session
+  SessionID         string
+  // The hostname of the worker that is executing the session
+  HostName          string
+  // ... other unexported fields
+}
+
+func GetSessionInfo(ctx Context) *SessionInfo
+```
+
+The Session Context also stores some session metadata, which can be retrieved by the `GetSessionInfo()` API.
+If the Context passed in doesn't contain any Session metadata, this API will return a `nil` pointer.
+
+### Concurrent Session Limitation
```

Comment:
```suggestion
### Concurrent Session limitation
```

Reply:
What do you think of renaming this header: "Limiting concurrent Sessions" or "Concurrent Sessions limitation" or "Limit concurrent Sessions"?